### DISCUSSION TITLE
[Art competition] Operational Semantics

### DISCUSSION THREAD

**Discussion by PSuHyeon** [prosyslab-classroom/cs424-program-reasoning#260] [Art competition]
![DALL·E 2022-11-30 22 48 22 - DALLE is artificial intelligent artist robot who draws picture defined in natural language in photorealistic style of art](https://user-images.githubusercontent.com/99390261/204812908-9d2ac2a2-3dab-4fb2-b763-c0d815d84dc0.png)

## Short Description
For me, [Operational Semantics] means [DALLE is artificial intelligent artist robot who draws picture defined in natural language in photorealistic style of art]".

## Long Description (up to 3 sentences)
Using operational semantics, semantics of the program is described by the semantics of its subcomponents or that of itself. 
In determining semantics of DALL-E, I used semantics of DALL-E to describe its semantics, thereby showing artistic way of explaining how operational semantics style can be use to define semantics of program!

**Comment by KihongHeo** [comment]
Wow. Recursion. What would be the fixed point (i.e., denotational semantics) of this? You should think of this because you also take CS524!

**Comment by PSuHyeon** [comment]
Unfortunately in this semantics, the domain would be infinite (i.e., DALL-E drawing DALL-E who draws DALL-E who draws DALL-E... ). 

Therefore, we must use widening operator to reach fixed point!

I would design result of widening as DALL-E drawing DALL-E who draws white blank (i.e., nothing). Any artistic work starts from and  expressed in the white blank, and white blank has potential to be any artistic works (i.e., narrowing!). Therefore, I think white blank is good ways to soundly subsume any artistic work (i.e., Top), which includes DALL-Es drawing each other.
